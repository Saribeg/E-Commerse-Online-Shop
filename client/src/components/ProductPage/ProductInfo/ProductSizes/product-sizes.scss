@import '../../../../scss/resources/resources';

.product-sizes {
  display: flex;
  justify-content: center;
  margin-bottom: 15px;
}

.size-item {
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid $grey-bg;
  padding: 16px;
  font-size: 14px;
  line-height: 20px;
  color: #515356;
  margin-right: 10px;
  cursor: pointer;

  &.active {
    border-color: #dedede;
    background-color: $grey-bg;
  }
}
