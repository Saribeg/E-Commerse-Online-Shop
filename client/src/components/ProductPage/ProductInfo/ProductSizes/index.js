import React, { Component, Fragment } from "react";
import ProductCounter from "../../../atomic/ProductCounter";
import './product-sizes.scss'

export default class ProductSizes extends Component {
  state = {
    productCount: 0
  };

  getAvailableSizes = maxcount => {
    this.setState({
      productCount: maxcount
    });
    console.log(this.state);
  };

  render() {
    let productFeatures = this.props.productFeatures;

    let sizes = productFeatures.map(elem => {
      if (elem.colorName === this.props.activeColor) {
        return elem.sizes.map(elem => {
          if (Number(elem.quantity) > 0) {
            return (
              <li key={elem._id}
                className="size-item"
                onClick={() => this.getAvailableSizes(elem.quantity)}
              >
                {elem.size}
              </li>
            );
          }
        });
      }
    });

    return (
      <Fragment>
        <p className="product-filter">Size</p>
        <ul className="product-sizes">{sizes}</ul>
        <p className="product-filter">Quantity</p>
        <ProductCounter maxCount={this.state.productCount} />
      </Fragment>
    );
  }
}
