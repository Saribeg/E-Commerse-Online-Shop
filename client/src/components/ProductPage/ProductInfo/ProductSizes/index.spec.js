import React from 'react';
import ProductSizes from './index';

const productFeatures = [{
	"color": "#000000",
	"colorName": "black",
	"imageUrls": [
		"/img/products/women/clothing/hoodies/002/000000/021.jpg",
		"/img/products/women/clothing/hoodies/002/000000/022.jpg",
		"/img/products/women/clothing/hoodies/002/000000/023.jpg",
		"/img/products/women/clothing/hoodies/002/000000/024.jpg"
	],
	"sizes": [{
			"size": "xs",
			"quantity": 0
		},
		{
			"size": "s",
			"quantity": 0
		},
		{
			"size": "m",
			"quantity": 18
		},

		{
			"size": "l",
			"quantity": 23
		}
	]
}];

describe('ProductSizes component', () => {
    it('ProductSizes component render', () => {
        const wrapper  = shallow(<ProductSizes productFeatures={productFeatures}/>)
        expect(wrapper).toMatchSnapshot()
    })
})