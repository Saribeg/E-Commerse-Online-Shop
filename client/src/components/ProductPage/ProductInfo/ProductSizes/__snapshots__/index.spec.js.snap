// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSizes component ProductSizes component render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductSizes
    productFeatures={
      Array [
        Object {
          "color": "#000000",
          "colorName": "black",
          "imageUrls": Array [
            "/img/products/women/clothing/hoodies/002/000000/021.jpg",
            "/img/products/women/clothing/hoodies/002/000000/022.jpg",
            "/img/products/women/clothing/hoodies/002/000000/023.jpg",
            "/img/products/women/clothing/hoodies/002/000000/024.jpg",
          ],
          "sizes": Array [
            Object {
              "quantity": 0,
              "size": "xs",
            },
            Object {
              "quantity": 0,
              "size": "s",
            },
            Object {
              "quantity": 18,
              "size": "m",
            },
            Object {
              "quantity": 23,
              "size": "l",
            },
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <p
          className="product-filter"
        >
          Size
        </p>,
        <ul
          className="product-sizes"
        />,
        <p
          className="product-filter"
        >
          Quantity Available: 
          0
        </p>,
        <ProductCounter
          maxCount={0}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Size",
          "className": "product-filter",
        },
        "ref": null,
        "rendered": "Size",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
          ],
          "className": "product-sizes",
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Quantity Available: ",
            0,
          ],
          "className": "product-filter",
        },
        "ref": null,
        "rendered": Array [
          "Quantity Available: ",
          0,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "maxCount": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <p
            className="product-filter"
          >
            Size
          </p>,
          <ul
            className="product-sizes"
          />,
          <p
            className="product-filter"
          >
            Quantity Available: 
            0
          </p>,
          <ProductCounter
            maxCount={0}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Size",
            "className": "product-filter",
          },
          "ref": null,
          "rendered": "Size",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
            ],
            "className": "product-sizes",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Quantity Available: ",
              0,
            ],
            "className": "product-filter",
          },
          "ref": null,
          "rendered": Array [
            "Quantity Available: ",
            0,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "maxCount": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
